### PR TITLE
Semantic colors: menu slots and filling in gaps in variant logic

### PR DESCRIPTION
**Pull request by bennettclark**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Adds new semantic slots, and adds variant logic for all existing slots I have design direction for. This should bring us up to date with all slots from variant/emphasis redlines that are approved at this time.

### CODE REVIEW

File: packages/variants/src/variants.ts [hunk 1]
```diff
@@ -170,7 +188,13 @@ export function getSoftVariant(theme: IPartialTheme): ITheme {
     // inputFocusBorderAlt: p.themePrimary,
     variantBorder: !fullTheme.isInverted ? p.neutralLight : p.neutralLighterAlt,
     variantBorderHovered: p.neutralTertiary,
-    defaultStateBackground: !fullTheme.isInverted ? p.themeLight : p.themeTertiary,
+    defaultStateBackground: !fullTheme.isInverted ? p.neutralLight : p.neutralQuaternaryAlt,
+
+    actionLink: p.neutralPrimary,
+    actionLinkHovered: !fullTheme.isInverted ? p.neutralDark : p.neutralPrimary,
```
Comment (phkuo) [inline packages/variants/src/variants.ts]: is this right? in inverted, the color doesn't change on hover?

File: packages/variants/src/variants.ts [hunk 2]
```diff
@@ -84,10 +84,19 @@ export function getNeutralVariant(theme: IPartialTheme): ITheme {
...
+    bodyText: p.neutralPrimary,
+    bodySubtext: p.neutralSecondary,
+    bodyDivider: p.neutralQuaternaryAlt,
     variantBorder: !fullTheme.isInverted ? p.neutralQuaternaryAlt : p.neutralLighterAlt,
     variantBorderHovered: p.neutralTertiary,
-    defaultStateBackground: p.neutralQuaternaryAlt,
+    defaultStateBackground: !fullTheme.isInverted ? p.neutralLight : p.neutralQuaternaryAlt,
+
+    actionLink: p.neutralPrimary,
+    actionLinkHovered: !fullTheme.isInverted ? p.neutralDark : p.neutralPrimary,
```
Comment (phkuo) [inline packages/variants/src/variants.ts]: same as below

File: packages/variants/src/variants.ts (hunk 1 above)
Comment (bennettclark) [reply packages/variants/src/variants.ts]: @phkuo Per the designer, I'm going to wait on this until a few things can get resolved on the design end. Apparently there is a lot of back and forth right now on link interactions. I'll close this PR and reopen once I have confirmation.

**Comment by bennettclark** [comment]
@phkuo Re-opened PR after receiving confirmation from @betrue-final-final that these are the correct values. Per Sirena "link/linkHovered and actionLink/actionLinkHovered values should be good. Hover colors don't change on dark themes because nothing is lighter than #fff and dark theme interaction states go lighter (seen in buttons and theme links)."

**Pull request merged**